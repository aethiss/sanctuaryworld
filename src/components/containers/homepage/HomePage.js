import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

// Components
import HomePageMobile from './HomePageMobile'
import HomePageDesktop from './HomePageDesktop'

@connect((state) => ({
  device: state.device,
  battletag: state.user?.battletag,
}), {})
class HomePage extends Component {

  render() {
    const { device, battletag } = this.props
    return (
      <>
        {device?.isPhone ?
          <HomePageMobile battletag={battletag} /> :
          <HomePageDesktop battletag={battletag} />
        }
      </>
    )
  }
}

HomePage.propTypes = {
  device: PropTypes.object,
  battletag: PropTypes.string,
}

export default HomePage
