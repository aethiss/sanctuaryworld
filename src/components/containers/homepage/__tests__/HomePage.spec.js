import React from 'react'

// Tester
import { Provider } from 'react-redux'
import TestRenderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
const mockStore = configureStore([])

// Component
import HomePage from '../HomePage'

const desktopProps = {
  device: 'desktop',
  isPhone: false,
}


describe('HomePage Component', () => {
  describe('Render', () => {
    it('[HomePage] Default render desktop', () => {
      const store = mockStore({
        device: desktopProps,
        battletag: 'aethiss',
      })
      const testRenderer = TestRenderer.create(
        <Provider store={store}>
          <HomePage />
        </Provider>,
      )
      expect(testRenderer).toMatchSnapshot()
    })
  })
})
