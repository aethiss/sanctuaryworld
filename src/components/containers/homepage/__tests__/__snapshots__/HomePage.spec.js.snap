// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component Render [HomePage] Default render desktop 1`] = `
<div
  style={
    Object {
      "backgroundImage": "url(\\"/images/background/desktop-home-background-ultrawide.jpg\\")",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "contain",
      "height": "100%",
      "position": "relative",
    }
  }
>
  <video
    autoPlay={true}
    muted={true}
    onEnded={[Function]}
    width="100%"
  >
    <source
      src="https://d3a5h34gwy5glx.cloudfront.net/assets/media/agegate-background-ultrawide-intro.9e333f400749237bc0aa377cb3664c90.mp4"
    />
  </video>
  <div
    style={
      Object {
        "left": "50%",
        "margin": 0,
        "position": "absolute",
        "textAlign": "center",
        "top": "70%",
        "transform": "translate(-50%, -50%)",
        "zIndex": 1,
      }
    }
  >
    <a
      href="/api/connect/bnet"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="makeStyles-button-1"
      >
        <span
          className="makeStyles-buttonLeft-3"
        />
        <span
          className="makeStyles-buttonMid-2"
        >
          <span
            className="makeStyles-buttonText-5"
          >
            Connect
          </span>
        </span>
        <span
          className="makeStyles-buttonRight-4"
        />
      </button>
    </a>
    <p>
      Secure Login with Blizzard Account
    </p>
  </div>
</div>
`;
